sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -551,10 +564,14 @@ struct SocRegistersImpl {
     #[register(offset = 0x062c, write_fn = on_write_internal_nmi_vector)]
     internal_nmi_vector: ReadWriteRegister<u32>,
 
-    /// INTERNAL_FW_UPDATE_RESET_WAIT_CYCLES Register
+    /// INTR_BLOCK_RF Register
     #[register_array(offset = 0x0800)]
     intr_block_rf: [u32; INTR_BLOCK_SIZE / 4],
 
+    /// ERROR_INTERNAL_INTR_R Register
+    #[register(offset = 0x0814)]
+    error_internal_intr_r: ReadWriteRegister<u32, ErrorIntrT::Register>,
+
```

Comment:
These registers overlap, so I'm not sure this will work as expected. I think instead, I think you could define something like

```rust
const INTR_BLOCK_ERROR_INTERNAL_INTR_R_OFFSET = 0x14;

// And then when you write to it, do something like
self.intr_block_rf[INTR_BLOCK_ERROR_INTERNAL_INTR_R_OFFSET] = <register_value>;
```

Reply:
Discussed and took a different approach of adding each register individually and removing intr_block_rf.